File: dsa_problems/minimizeLoss.cpp
#include <bits/stdc++.h>
using namespace std;

void cntLoss(vector<int> &price) {
    int minLoss = INT_MAX;
    int buyYear = -1, sellYear = -1;
    int n = price.size();
    
    for (int buy = 0; buy < n - 1; buy++) {
        for (int sell = buy + 1; sell < n; sell++) {
            if (price[sell] < price[buy]) {
                int loss = price[buy] - price[sell];
                if (loss < minLoss) {
                    minLoss = loss;
                    buyYear = buy + 1;
                    sellYear = sell + 1;
                }
            }
        }
    }
    
    if (buyYear == -1) {
        cout << "No valid loss found" << endl;
    } else {
        cout << "Buy in year " << buyYear << " and sell in year " << sellYear << " with a loss of " << minLoss << endl;
    }
}

int main() {
    int n; 
    cout << "Enter total number of year: "; 
    cin >> n; 
    
    vector<int> price(n); 
    cout << "Enter price: "; 
    for (int i = 0; i < n; i++) {
         cin >> price[i]; 
    }
    cntLoss(price); 
    return 0;
}

File: dsa_problems/minimizeLoss.py

def cnt_loss(price):
    min_loss = float('inf')
    buy_year, sell_year = -1, -1
    n = len(price)
    
    for buy in range(n - 1):
        for sell in range(buy + 1, n):
            if price[sell] < price[buy]:
                loss = price[buy] - price[sell]
                if loss < min_loss:
                    min_loss = loss
                    buy_year = buy + 1
                    sell_year = sell + 1
    
    if buy_year == -1:
        print("No valid loss found")
    else:
        print(f"Buy in year {buy_year} and sell in year {sell_year} with a loss of {min_loss}")


n = int(input("Enter total number of years: "))
price = list(map(int, input("Enter prices: ").split()))
cnt_loss(price)